core/src/main/scala/org/typelevel/otel4s/trace/SamplingDecision.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.trace
+
+/** A decision on whether a span should be recorded or dropped.
+  */
+sealed abstract class SamplingDecision(val isSampled: Boolean)
```

Comment:
In the Java client there is a RECORD_ONLY type as well for client based sampling it looks like.  Is that something we wanted to have here?  https://github.com/open-telemetry/opentelemetry-java/blob/50408d499f85d5761d0a5ed9bf9d77d5ff01fff5/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/SamplingDecision.java#L17

Reply:
I would like to mirror all types, but I do not understand how to construct the proper type since the input type is `Boolean`:

https://github.com/open-telemetry/opentelemetry-java/blob/86eea94b7695aba4a871e04f042fd0c0e84f607f/api/all/src/main/java/io/opentelemetry/api/trace/SpanContext.java#L126

OpenTelemetry calculates `SamplingDecision` in internals:
https://github.com/open-telemetry/opentelemetry-java/blob/50408d499f85d5761d0a5ed9bf9d77d5ff01fff5/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java#L190-L195

Reply:
I'll take a look at those.